packages/search/src/components/InputSearch/index.tsx

```diff
@@ -0,0 +1,70 @@
+import React, { useLayoutEffect, useRef, useState } from 'react';
+import { MagnifyingGlass } from '@rainbow-modules/icons';
+import { InputComponent, InputSearchProps } from './types';
+import { StyledContainer, StyledInput } from './styled';
+import Trailing from './trailing';
+
+const InputSearch: React.FC<InputSearchProps> = ({
+    className,
+    style,
+    disabled,
+    readOnly,
+    name,
+    value,
+    variant,
+    onChange,
+    onSearch,
+}: InputSearchProps) => {
+    const trailingRef = useRef<HTMLDivElement>(null);
+    const inputRef = useRef<InputComponent>(null);
+    const [paddingRight, setPaddingRight] = useState<number>();
+
+    useLayoutEffect(() => {
+        if (trailingRef.current) {
+            const { width } = trailingRef.current.getBoundingClientRect();
+            if (paddingRight !== width) setPaddingRight(width);
+        }
+    }, [paddingRight, value]);
```

Comment:
I think we can compute in a functional way the padding needed based on the props (value and variant) and then avoid using a state for the padding